src/main/java/randoop/output/FailingTestFilter.java

```diff
@@ -0,0 +1,316 @@
+   * @param javaFileWriter the {@link JavaFileWriter} to write {@code .java} files for the classes
+   */
+  public FailingTestFilter(TestEnvironment testEnvironment, JavaFileWriter javaFileWriter) {
+    this.testEnvironment = testEnvironment;
+    this.javaFileWriter = javaFileWriter;
+    this.classLoader = new SequenceClassLoader(getClass().getClassLoader());
+  }
+
+  /**
+   * {@inheritDoc}
+   *
+   * <p>Inserts comments in place of assertions that fail when the test is run.
+   *
+   * <p>Assumes output from JUnit4 {@code org.junit.runner.JUnitCore} runner used in {@link
+   * TestEnvironment}.
+   */
+  @Override
+  public File writeClassCode(String packageName, String classname, String classSource) {
+
+    String qualifiedClassname = (packageName.isEmpty() ? "" : packageName + ".") + classname;
+
+    int pass = 0; // Used to create unique working directory name.
+    boolean passing = false;
+
+    while (!passing) {
+      Path workingDirectory = createWorkingDirectory(classname, pass);
+
+      compileTestClass(packageName, classname, classSource, workingDirectory);
+
+      RunCommand.Status status;
+      try {
+        status = testEnvironment.runTest(qualifiedClassname, workingDirectory.toFile());
+      } catch (ProcessException e) {
+        throw new BugInRandoopException("Error filtering regression tests", e);
+      }
+
+      if (status.exitStatus == 0) {
+        passing = true;
+      } else {
+        classSource = commentFailingAssertions(classname, classSource, status);
+      }
+      pass++;
+    }
+    return javaFileWriter.writeClassCode(packageName, classname, classSource);
+  }
+
+  @Override
+  public File writeUnmodifiedClassLines(
+      String packageName, String classname, String[] sourceLines) {
+    return javaFileWriter.writeUnmodifiedClassLines(packageName, classname, sourceLines);
+  }
+
+  @Override
+  public File writeUnmodifiedClassCode(String packageName, String classname, String classCode) {
+    return javaFileWriter.writeClassCode(packageName, classname, classCode);
+  }
+
+  /**
+   * Uses the failures in the {@code status} from running JUnit with {@code classCode} to identify
+   * lines with failing assertions and replaces them with a line comment with the assertion text
+   *
+   * @param classname the name of the test class
+   * @param classCode the source code for the test class, each assertion must be on its own line
+   * @param status the {@link RunCommand.Status} for running the test with JUnit
+   * @return the class source edited so that failing assertions are replaced by line comments
+   * @throws BugInRandoopException if {@code status} contains output for a failure not involving a
+   *     Randoop generated test method
+   */
+  private String commentFailingAssertions(
+      String classname, String classCode, RunCommand.Status status) {
+    // JUnit4 writes to standard out, check this doesn't change.
+    assert status.errorOutputLines.isEmpty()
```

Comment:
Here and elsewhere, I'm worried about silently ignoring errors when assertions are turned off.
Wouldn't it be better to throw the error unconditionally so that we get bug reports from users and we fix the problems?